### ISSUE TITLE
Exclude table ID from search.

### ISSUE THREAD

**Issue by rgiaviti** [yajra/laravel-datatables-oracle#13]
Hi!

I realized a strange behavior when searching. When I search using the search field in datatable, the column ID is using in query. Take a look:

```sql
select *
          from po_base_rateio
         inner join po_anos
            on po_anos.id = po_base_rateio.ano_id
         inner join po_direcionadores
            on po_direcionadores.id = po_base_rateio.direcionador_id
         inner join po_atividades
            on po_atividades.id = po_base_rateio.atividade_id
         where (id LIKE '%1034%' or po_anos.ano LIKE '%1034%' or
               po_direcionadores.descricao LIKE '%1034%' or
               po_atividades.codigo LIKE '%1034%' or an8 LIKE '%1034%' or oferta LIKE '%1034%' or
               os LIKE '%1034%' or janeiro LIKE '%1034%' or fevereiro LIKE '%1034%' or
               marco LIKE '%1034%' or abril LIKE '%1034%' or maio LIKE '%1034%' or junho LIKE '%1034%' or
               julho LIKE '%1034%' or agosto LIKE '%1034%' or setembro LIKE '%1034%' or
               outubro LIKE '%1034%' or novembro LIKE '%1034%' or dezembro LIKE '%1034%')
         order by po_anos.ano desc, po_base_rateio.direcionador_id asc;
```

At where clause, the query is using an ID to search, but I don't want to search using ID.
```id LIKE '%1034%'```

My view:

```javascript
    	var table = $('#basesRateioTable').DataTable({
            "serverSide": true,
            "processing": true,
            "ajax": "{!! URL::action('Admin\BaseRateioController@consulta') !!}",
            "columns": [    
                {data: 'ano', name: 'anos.ano'},
                {data: 'descricao', name: 'direcionadores.descricao'},
                {data: 'codigo', name: 'atividades.codigo'},
                {data: 'an8', name: 'an8'},
                {data: 'oferta', name: 'oferta'},
                {data: 'os', name: 'os'},
                {data: 'janeiro', name: 'janeiro'},
                {data: 'fevereiro', name: 'fevereiro'},
                {data: 'marco', name: 'marco'},
                {data: 'abril', name: 'abril'},
                {data: 'maio', name: 'maio'},
                {data: 'junho', name: 'junho'},
                {data: 'julho', name: 'julho'},
                {data: 'agosto', name: 'agosto'},
                {data: 'setembro', name: 'setembro'},
                {data: 'outubro', name: 'outubro'},
                {data: 'novembro', name: 'novembro'},
                {data: 'dezembro', name: 'dezembro'}
            ],            
            "language": {
                "url": "{!! URL::to('plugins/datatable/js/Portuguese.json') !!}"
            }
        });
```

And my Controller:

```php
            $colunas = ['base_rateio.id','anos.ano', 'direcionadores.descricao', 'atividades.codigo', 'base_rateio.an8', 'base_rateio.oferta',
                        'base_rateio.os', 'base_rateio.janeiro', 'base_rateio.fevereiro', 'base_rateio.marco', 'base_rateio.abril',
                        'base_rateio.maio', 'base_rateio.junho', 'base_rateio.julho', 'base_rateio.agosto', 'base_rateio.setembro',
                        'base_rateio.outubro', 'base_rateio.novembro', 'base_rateio.dezembro'];

            $basesRateio = BaseRateio::getBaseRateioOrdenadoAno()->select($colunas);
            return Datatables::of($basesRateio)->make(true);
```

The search in my Model:

```php
		return BaseRateio::join('anos', 'anos.id', '=', 'base_rateio.ano_id')
                         ->join('direcionadores', 'direcionadores.id', '=', 'base_rateio.direcionador_id')
                         ->join('atividades', 'atividades.id', '=', 'base_rateio.atividade_id')
                         ->orderBy('anos.ano', 'desc')
						 ->orderBy('base_rateio.direcionador_id', 'asc');
```

I don't know if this is bug, because I am not specifying the ID column anywhere, and the component still use this column as search parameter.

**Comment by yajra** [comment]
thanks for reporting!

**Comment by rgiaviti** [comment]
I'm glad that I can help. If you need more help testing and reporting, I can do that. Thanks!

**Comment by rgiaviti** [comment]
Hi! I just update the laravel-datables to version 4.0.8 and another strange behavior appeared. When I search in datatable, the generated query is wrong:

```sql
select *
 from po_vw_dt_base_rateio
 where (LIKE '%1034%' or id LIKE '%1034%' or ano LIKE '%1034%' or
        descricao LIKE '%1034%' or codigo LIKE '%1034%' or
        codigo_cliente LIKE '%1034%' or oferta LIKE '%1034%' or
        os LIKE '%1034%' or janeiro LIKE '%1034%' or
        fevereiro LIKE '%1034%' or marco LIKE '%1034%' or
        abril LIKE '%1034%' or maio LIKE '%1034%' or
        junho LIKE '%1034%' or julho LIKE '%1034%' or
        agosto LIKE '%1034%' or setembro LIKE '%1034%' or
        outubro LIKE '%1034%' or novembro LIKE '%1034%' or
        dezembro LIKE '%1034%')
```

In this line:
```sql
 where (LIKE '%1034%' or id LIKE '%1034%' or ano LIKE '%1034%' or
```

Is missing an expression right after where clause. Another thing, is ID is used on query.

**Comment by yajra** [comment]
Hi! I can't reproduce the same error like you posted. Can you please provide more details like the view and the controller code?

**Comment by rgiaviti** [comment]
Sure!

Starting with route:
```php
$router->get('premissa/consulta', 'Admin\PremissaController@consulta');
```

Controller:
```php
    public function consulta(Request $request) {
        if ($request->ajax()) {
            $colunas = ['id','ano', 'direcionador', 'calculo', 'conta', 'descricao', 'valor'];

            $premissas = DB::table('vw_dt_premissas')->select($colunas);
            return Datatables::of($premissas)->make(true);
        }
        else {
            App::abort(400, 'Bad Request');
        }
    }
```
The "vw_dt_premissas" is a Oracle view.

My View:
```javascript
$('#premissasTable').DataTable({
        "serverSide": true,
        "processing": true,
        "ajax": "{!! URL::action('Admin\PremissaController@consulta') !!}",
        "columns": [
            {
                "orderable":      false,
                "data":           null,
                "defaultContent": "<a href='#' id='edit' role='button' class='btn btn-primary btn-sm pull-left' data-toggle='tooltip' data-placement='top' data-original-title='Editar'><i class='glyphicon glyphicon-pencil'></i></a>"
            },
            {data: 'id', name: 'id'},
            {data: 'ano', name: 'ano'},
            {data: 'direcionador', name: 'direcionador'},
            {data: 'calculo', name: 'calculo'},
            {data: 'conta', name: 'conta'},
            {data: 'descricao', name: 'descricao'},
            {data: 'valor', name: 'valor'}
        ],
        "language": {
            "url": "{!! URL::to('plugins/datatable/js/Portuguese.json') !!}"
        }
    });
```

The versions:
 * Laravel Datatables: 4.0.8;
 * Datatables: 1.10.5

**Comment by rgiaviti** [comment]
Hi! Just found the problem. When I configured the first column in Javascript:

```javascript
            {
                "orderable":      false,
                "data":           null,
                "defaultContent": "<a href='#' id='edit' role='button' class='btn btn-primary btn-sm pull-left' data-toggle='tooltip' data-placement='top' data-original-title='Editar'><i class='glyphicon glyphicon-pencil'></i></a>"
            },
```

There's no `name` defined. So, the the query is generated a blank identifier is used causing the problem. The solution is simple. Define a name or exclude from search using attribute `searchable`. Here an example:

```javascript
				{
                    searchable:     false,
				    orderable:      false,
				    data:           null,
				    defaultContent: "<a href='#' id='edit' role='button' class='btn btn-primary btn-sm pull-left' data-toggle='tooltip' data-placement='top' data-original-title='Editar'><i class='glyphicon glyphicon-pencil'></i></a>"
                },
```

Or you can define a name for column:
```javascript
                {
                   searchable: false,
                   orderable: false,
                   visible: false,
                   data: 'id',
                   name: 'id'
                },
```

Thanks again!

**Comment by yajra** [comment]
Yeah your right. I think you can leave the name blank but have to set it as `searchable: false` since it was set as `true` by default and it triggers the search function but did not found the appropriate column name.. Will try to capture that exception tom. Thanks for the info.

**Comment by yajra** [comment]
searching without name should be fixed now on v4.0.10.. https://github.com/yajra/laravel-datatables-oracle/blob/master/src/yajra/Datatables/Datatables.php#L336

**Comment by rgiaviti** [comment]
Awesome work! Thanks
